### PR TITLE
Problem: relayer should only relay supported tokens

### PR DESCRIPTION
**Pull request by thomas-nguy**

### CODE REVIEW

File: orchestrator/relayer/src/batch_relaying.rs [hunk 1]
```diff
@@ -74,14 +75,28 @@ async fn get_batches_and_signatures(
     current_valset: Valset,
     grpc_client: &mut GravityQueryClient<Channel>,
     gravity_id: String,
+    supported_contracts: Vec<EthAddress>,
 ) -> HashMap<EthAddress, Vec<SubmittableBatch>> {
-    let latest_batches = if let Ok(lb) = get_latest_transaction_batches(grpc_client).await {
-        lb
+    let mut latest_batches: Vec<TransactionBatch> = Vec::new();
+    if supported_contracts.len() == 0 {
```
Comment (tomtau) [inline orchestrator/relayer/src/batch_relaying.rs]: nit: could be `supported_contracts.is_empty()`?

File: orchestrator/relayer/src/batch_relaying.rs [hunk 2]
```diff
@@ -74,14 +75,28 @@ async fn get_batches_and_signatures(
...
     } else {
-        return HashMap::new();
-    };
+        // apply a whitelist to only supported contracts
+        for contract in supported_contracts {
+            if let Ok(tb) = get_latest_batch(grpc_client, contract).await {
+                latest_batches.push(tb);
+            }
+        }
+        if latest_batches.len() == 0 {
```
Comment (tomtau) [inline orchestrator/relayer/src/batch_relaying.rs]: ditto

**Review by tomtau — COMMENTED** [review]
does the integration test use the supported tokens config? perhaps it could be extended with additional assertions to check unsupported tokens weren't relayed?

File: orchestrator/relayer/src/batch_relaying.rs (hunk 1 above)
Comment (thomas-nguy) [reply orchestrator/relayer/src/batch_relaying.rs]: done

File: orchestrator/relayer/src/batch_relaying.rs (hunk 2 above)
Comment (thomas-nguy) [reply orchestrator/relayer/src/batch_relaying.rs]: done

**Comment by thomas-nguy** [comment]
I have tested manually but I couldnt change the integration tests because it is a lot of work (contract addresses need to be deployed and configured beforehand before starting relayer, but now, we are deploying the contract during the test)

**Review by tomtau — APPROVED** [review]
can you open an issue to extend the integration test? @adu-crypto may be able to help after wrapping up the simulations?

**Comment by thomas-nguy** [comment]
sure

**Pull request merged**